src/c++/simulator/materials.hpp

```diff
@@ -0,0 +1,22 @@
+﻿#pragma once
+
+#include <simulator/my_types.hpp>
+
+struct IMaterial
+{
+public:
+	virtual Material get_type() = 0;
+	double get_density() const;
+
+public:
```

Comment:
I would add those members as `protected` and add `get_young_modulus` and `get_possion_ratio` as `public`  getters.